src/main/java/org/commcare/util/EncryptionUtils.java

```diff
@@ -92,16 +220,25 @@ public static String decrypt(String message, String key) throws EncryptionExcept
             bb.get(cipherText);
 
 
-            Cipher cipher = Cipher.getInstance(ENCRYPT_ALGO);
-            cipher.init(Cipher.DECRYPT_MODE, secret, new GCMParameterSpec(TAG_LENGTH_BIT, iv));
+            Cipher cipher = Cipher.getInstance(getCryptographicTransformation(algorithm));
+            if (algorithm.equals("AES")) {
+                cipher.init(Cipher.DECRYPT_MODE, key, new GCMParameterSpec(TAG_LENGTH_BIT, iv));
+            } else {
+                cipher.init(Cipher.DECRYPT_MODE, key);
+            }
             byte[] plainText = cipher.doFinal(cipherText);
             return new String(plainText, Charset.forName("UTF-8"));
         } catch (NoSuchAlgorithmException | BadPaddingException | NoSuchPaddingException |
-                IllegalBlockSizeException | InvalidAlgorithmParameterException | InvalidKeyException | Base64DecoderException e) {
+                 IllegalBlockSizeException | InvalidKeyException | Base64DecoderException |
+                 InvalidAlgorithmParameterException e) {
             throw new EncryptionException("Error encountered while decrypting the message", e);
         }
     }
 
+    public static boolean isAndroidKeyStoreSupported() {
```

Comment:
Should be in `commcare-android`

Reply:
What about giving it a generic name (like `isKeyStoreSupported` or `isPlatformKeyStoreAvailable`) and making it part of the CommCare platform classes?

Reply:
99e70fdf6c8533ca1e679bc96c7363b7b51db941. The method was renamed to `isPlatformKeyStoreAvailable`.